timeboost-sequencer/src/lib.rs

```diff
@@ -350,13 +356,19 @@ impl Task {
         let mut candidates = Vec::new();
         while !actions.is_empty() {
             let mut round = RoundNumber::genesis();
+            let mut evidence = Evidence::Genesis;
             let mut lists = Vec::new();
             while let Some(action) = actions.pop_front() {
                 match action {
                     Action::Deliver(payload) => {
                         round = payload.round();
```

Comment:
Can this update also be moved into the `Ok(data)` branch? I think we should either use all or none of the payload values.